### PR TITLE
Add k8s and CF TaskLaunchers into a single server implementation

### LINKED ISSUE #2627: Add k8s and CF TaskLaunchers into a single server implementation
Adapt local server to use multiple local TaskLauncher implementationsThis would be us following the similar model that we do in Skipper. That would mean, we would pack the single-binary with local, cf, and k8s-deployer libraries and add support for "multiple" platform profiles through configuration map.

Use Skipper's existing multi-platform strategy as a reference. Platform-specific configurations will be a map of config overrides.

child of #2613

### PR DESCRIPTION
**Pull request by markpollack**
- Renamed spring-cloud-dataflow-server-local to spring-cloud-dataflow-server
- Two new maven modules that define platform releated Beans
- Disabled CF security until update to boot 2.1 + security is completed
- Default version of local task deployer is created only if no other task deployers defined

Note

- TaskExecutionControllerTests is failing from the maven build - not in IDE, don't know why.  Tried a few things to no avail.
- Used a 'local' profile logback-scdf-local.xml to avoid local file logging on cloud platforms
- LocalSchedulerAutoConfiguration uses 

```
@Profile("!cloud") // Server not deployed on CF
@ConditionalOnProperty(name = "kubernetes.service.host", matchIfMissing = true) // Server not deployed on K8s
```

as conditions to create the local server implementation.   Was hoping to avoid having to specify a 'local' profile.  There maybe better ways more open to extension for other cloud platforms.

Limitations

- Can only configure a single deployer per platform and it needs to be named 'default'

Tested on local and on CF using the following manifest. **Did not test on k8s**

I believe one can't use the format

```
 SPRING_CLOUD_DATAFLOW_TASK_PLATFORM_CLOUDFOUNDRY_ACCOUNTS[default]_CONNECTION_URL: https://api.sys.huron.cf-app.com
```
to specify the accounts, since boot is more strict with map of maps names in environment variable.  Used SPRING_APPLICATION_JSON instead, which isn't as clean looking.

```
applications:
- name: mlp-dataflow-server
  memory: 1G
  disk_quota: 2G
  buildpack: java_buildpack_offline
  timeout: 180
  path: ./spring-cloud-dataflow-server-2.0.0.BUILD-SNAPSHOT.jar
  services:
    - mysql
  env:
    SPRING_APPLICATION_NAME: data-flow-server
    SPRING_APPLICATION_JSON: '{ "maven": { "remote-repositories": { "repo1": { "url": "https://repo.spring.io/libs-snapshot" } } }, "spring": { "cloud": { "dataflow": { "task": { "platform": { "cloudfoundry": { "accounts": { "default": { "connection": { "url": "https://api.sys.huron.cf-app.com", "org": "mpollack", "space": "test", "domain": "app.huron.cf-app.com", "username": "***", "password": "****", "skipSslValidation": true }, "deployment": { "memory": "2048m", "disk": "2048m", "services": "mysql", "deleteRoutes": false } } } } } } } } } }'
    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_TASK_API_TIMEOUT: 300
    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_TASK_SERVICES: mysql
    TRUST_CERTS: api.sys.huron.cf-app.com
    SPRING_CLOUD_SKIPPER_CLIENT_SERVER_URI: http://localhost:7577/api
    SPRING_CLOUD_DATAFLOW_APPLICATIONPROPERTIES_STREAM_TRUST_CERTS: api.sys.huron.cf-app.com
    SPRING_CLOUD_DATAFLOW_APPLICATIONPROPERTIES_TASK_TRUST_CERTS: api.sys.huron.cf-app.com
    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_STREAM_ENABLE_RANDOM_APP_NAME_PREFIX: false
    SPRING_CLOUD_DATAFLOW_FEATURES_STREAMS_ENABLED: false
    SPRING_CLOUD_DATAFLOW_FEATURES_ANALYTICS_ENABLED: false
```

Fixes #2627

### CODE REVIEW

**Comment by chrisjs** [comment]
so far i was able to do some functional testing:

i was able to get ticktock running with the following changes applied:

* https://github.com/markpollack/spring-cloud-dataflow/pull/2 <-- branched from this PR branch and can be sucked in
* https://github.com/spring-cloud/spring-cloud-dataflow-server-kubernetes/pull/372 <-- should wait to be merged until these PR's are in master

running tasks failed with missing batch tables though:

```java.lang.IllegalStateException: Failed to execute CommandLineRunner
    at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:816) [spring-boot-2.1.0.RELEASE.jar!/:2.1.0.RELEASE]
    at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:797) [spring-boot-2.1.0.RELEASE.jar!/:2.1.0.RELEASE]
    at org.springframework.boot.SpringApplication.run(SpringApplication.java:324) [spring-boot-2.1.0.RELEASE.jar!/:2.1.0.RELEASE]
    at org.springframework.boot.SpringApplication.run(SpringApplication.java:1260) [spring-boot-2.1.0.RELEASE.jar!/:2.1.0.RELEASE]
    at org.springframework.boot.SpringApplication.run(SpringApplication.java:1248) [spring-boot-2.1.0.RELEASE.jar!/:2.1.0.RELEASE]
    at org.springframework.cloud.task.app.timestamp.batch.TimestampBatchTaskApplication.main(TimestampBatchTaskApplication.java:29) [classes!/:2.1.0.BUILD-SNAPSHOT]
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_192]
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_192]
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_192]
    at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_192]
    at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48) [timestamp-batch-task.jar:2.1.0.BUILD-SNAPSHOT]
    at org.springframework.boot.loader.Launcher.launch(Launcher.java:87) [timestamp-batch-task.jar:2.1.0.BUILD-SNAPSHOT]
    at org.springframework.boot.loader.Launcher.launch(Launcher.java:50) [timestamp-batch-task.jar:2.1.0.BUILD-SNAPSHOT]
    at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:51) [timestamp-batch-task.jar:2.1.0.BUILD-SNAPSHOT]
Caused by: org.springframework.jdbc.BadSqlGrammarException: PreparedStatementCallback; bad SQL grammar [SELECT JOB_INSTANCE_ID, JOB_NAME from BATCH_JOB_INSTANCE where JOB_NAME = ? and JOB_KEY = ?]; nested exception is java.sql.SQLSyntaxErrorException: (conn=11) Table 'mysql.BATCH_JOB_INSTANCE' doesn't exist```

in the server startup, looks like sql files for batch weren't being executed:

```2018-12-13 21:07:12.921  INFO 1 --- [           main] o.s.c.d.s.r.s.DataflowRdbmsInitializer   : Adding dataflow schema classpath:schemas/mysql/common.sql for mysql database
2018-12-13 21:07:12.973  INFO 1 --- [           main] o.s.c.d.s.r.s.DataflowRdbmsInitializer   : Adding dataflow schema classpath:schemas/mysql/streams.sql for mysql database
2018-12-13 21:07:12.974  INFO 1 --- [           main] o.s.c.d.s.r.s.DataflowRdbmsInitializer   : Adding dataflow schema classpath:schemas/mysql/tasks.sql for mysql database
2018-12-13 21:07:12.974  INFO 1 --- [           main] o.s.c.d.s.r.s.DataflowRdbmsInitializer   : Adding dataflow schema classpath:schemas/mysql/deployment.sql for mysql database
2018-12-13 21:07:12.975  INFO 1 --- [           main] o.s.c.d.s.r.s.DataflowRdbmsInitializer   : Adding dataflow schema classpath:schemas/mysql/jpa.sql for mysql database```

**Comment by chrisjs** [comment]
in regards to https://github.com/spring-cloud/spring-cloud-dataflow-server-kubernetes/pull/372/files - seems like we should probably copy those config files under `src` into the main data flow since the servers will be consolidated, otherwise we won't have deployment files

**Comment by markpollack** [comment]
@chrisjs I believe some additional pom deps for k8s-platform were needed..right?

**Comment by chrisjs** [comment]
@markpollack yes its in one of the PR’s listed above (https://github.com/markpollack/spring-cloud-dataflow/pull/2)

File: spring-cloud-dataflow-platform-cloudfoundry/src/main/java/org/springframework/cloud/dataflow/server/config/cloudfoundry/CloudFoundryDataFlowServerConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import org.springframework.cloud.deployer.spi.cloudfoundry.CloudFoundryConnectionProperties;
+import org.springframework.cloud.deployer.spi.cloudfoundry.CloudFoundryDeploymentProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Configuration class for customizing Cloud Foundry deployer.
+ *
+ * @author Eric Bottard
+ */
```
[the remark this answers is not in the document]
Comment (markpollack) [reply spring-cloud-dataflow-platform-cloudfoundry/src/main/java/org/springframework/cloud/dataflow/server/config/cloudfoundry/CloudFoundryDataFlowServerConfiguration.java]: We would have to add that to all the classes copied over from the scdf-cf project - they really aren't "new" per-se.

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskJobService.java [hunk 2]
```diff
@@ -218,7 +218,8 @@ public void restartJobExecution(long jobExecutionId) throws NoSuchJobExecutionEx
 		TaskExecution taskExecution = this.taskExplorer.getTaskExecution(taskJobExecution.getTaskId());
 		TaskDefinition taskDefinition = this.taskDefinitionRepository.findById(taskExecution.getTaskName())
 				.orElseThrow(() -> new NoSuchTaskDefinitionException(taskExecution.getTaskName()));
-		taskService.executeTask(taskDefinition.getName(), taskDefinition.getProperties(), taskExecution.getArguments());
+		taskService.executeTask(taskDefinition.getName(), taskDefinition.getProperties(), taskExecution.getArguments(),
+				"default");
```
[the remark this answers is not in the document]
Comment (markpollack) [reply spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskJobService.java]: `default` is the TaskDeployer to use if nothing is specified, so when on CF, likely default is the CF TaskDeployer impl and so on.

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskService.java [hunk 3]
```diff
@@ -370,7 +373,7 @@ private void destroyChildTask(String name) {
 	}
 
 	private void destroyTask(TaskDefinition taskDefinition) {
-		// TODO GH-2616 cleanup
+		// TODO GH-
```
[the remark this answers is not in the document]
Comment (markpollack) [reply spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskService.java]: GH-2678

**Comment by markpollack** [comment]
Rebased and incorporated changes suggested here, made a new PR - #2688

**Pull request closed without merging**